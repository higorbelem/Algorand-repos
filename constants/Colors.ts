const tintColorDark = '#fff';

export const Colors = {
  text: '#FFFFFF',
  background: '#191C21',
  tint: tintColorDark,
  icon: '#9BA1A6',
  tabIconDefault: '#9BA1A6',
  tabIconSelected: tintColorDark,
  main: '#2D2DF1',
  green: '#238636'
};
