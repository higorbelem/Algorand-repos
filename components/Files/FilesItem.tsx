import { StyleSheet, TouchableOpacity } from "react-native"

import { repoContent } from "@/@types/github";
import FolderIcon from "@/assets/svgs/folder.svg";
import FileIcon from "@/assets/svgs/file.svg";
import { Colors } from "@/constants/Colors";

import { Text } from "../Text";

export type FilesItemProps = repoContent & {
    isFirst: boolean;
   onPress: () => void;
};

export const FilesItem = (props: FilesItemProps) => {
    const style = styles(props.isFirst);

    return(
        <TouchableOpacity style={style.container} activeOpacity={0.7} onPress={props.onPress}>
            {
                props.type === 'dir' ? (
                    <FolderIcon width={20} height={20}/>
                ) : (
                    <FileIcon width={20} height={20}/>
                )
            }
            <Text>{props.name}</Text>
        </TouchableOpacity>
    )
}

const styles = (isFirst: boolean) => StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        gap: 10,
        padding: 15,
        width: '100%',
        borderTopWidth: isFirst ? 0 : 1,
        borderTopColor: Colors.text
    },
    image: {
        width: 20,
        height: 20,
        borderRadius: 5
    }
});