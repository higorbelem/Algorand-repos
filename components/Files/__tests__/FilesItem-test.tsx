import * as React from 'react';
import { render } from '@testing-library/react-native';

import { reposContentMock } from '@/constants/mocks';

import { FilesItem } from '../FilesItem';

const mockData = reposContentMock[0];

describe('FilesItem component', () => {
  it('Renders correctly', () => {
    const tree = render(<FilesItem {...mockData} isFirst onPress={() => {}}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should show correct text', () => {
    const tree = render(<FilesItem {...mockData} isFirst onPress={() => {}}/>);

    const component = tree.getByText(mockData.name);

    expect(component).toBeTruthy();
  });

  it('Should show folder icon', () => {
    const tree = render(<FilesItem {...mockData} type='dir' isFirst onPress={() => {}}/>);

    const component = tree.getByTestId("files-item-folder-icon");

    expect(component).toBeTruthy();
  });

  it('Should show folder icon', () => {
    const tree = render(<FilesItem {...mockData} type='file' isFirst onPress={() => {}}/>);

    const component = tree.getByTestId("files-item-file-icon");

    expect(component).toBeTruthy();
  });
});