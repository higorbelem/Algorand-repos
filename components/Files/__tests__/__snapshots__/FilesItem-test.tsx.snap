// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilesItem component Renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderTopColor": "#FFFFFF",
      "borderTopWidth": 0,
      "flex": 1,
      "flexDirection": "row",
      "gap": 10,
      "opacity": 1,
      "padding": 15,
      "width": "100%",
    }
  }
  testID="files-item"
>
  <svg
    height={20}
    testID="files-item-folder-icon"
    width={20}
  />
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
        },
        undefined,
        {
          "fontSize": 16,
          "fontWeight": "400",
        },
      ]
    }
  >
    .ci
  </Text>
</View>
`;
